src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -1003,7 +1070,11 @@ protected void CheckFeatureAvailabilityAndRuntimeSupport(SyntaxNode declarationS
         /// If the method has both block body and an expression body
         /// present, this is not treated as expression-bodied.
         /// </remarks>
-        internal abstract bool IsExpressionBodied { get; }
+        internal bool IsExpressionBodied => flags.IsExpressionBodied;
+        protected bool HasAnyBody => flags.HasAnyBody;
+
+        public sealed override RefKind RefKind => this.flags.RefKind;
+        public sealed override bool IsVararg => this.flags.IsVararg;
```

Comment:
sealed these to ensure no base types are doing this work through some other means.  all base types need to store this info in the Flags value.